client/engine/messageservice/p2p-message-service/service.go

```diff
@@ -164,6 +242,20 @@ func (s *P2PMessageService) Out() <-chan protocols.Message {
 
 // Close closes the P2PMessageService
 func (s *P2PMessageService) Close() error {
+	s.cancel()
+	s.mdns.Close()
 	s.p2pHost.RemoveStreamHandler(PROTOCOL_ID)
 	return s.p2pHost.Close()
 }
+
+// PeerInfoReceived returns a channel that receives a PeerInfo when a peer is discovered
+func (s *P2PMessageService) PeerInfoReceived() <-chan PeerInfo {
```

Comment:
Since we don't know exactly when the peers get added to a message service, `PeerInfoReceived` is very useful to wait until everyone has all the peer info in tests.